src/main/java/reactor/kafka/receiver/internals/DefaultKafkaReceiver.java

```diff
@@ -257,7 +256,8 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
                         if (requestsPending.get() > 0)
                             pollEvent.scheduleIfRequired();
                     })
-                .doOnCancel(() -> cancel(true));
+                .doOnTerminate(this::dispose)
```

Comment:
@smaldini in my code, I often use `doFinally` for this use case.
Is it valid, or `doOnTerminate` + `doOnCancel` is better?

Reply:
Do Finally has different semantic. If first emit signal to downstream and then execute handler where doOnCancel and doOnTerminal exexutes handler first and then send propagate signal to downstream